x-pack/plugin/enterprise-search/src/main/java/org/elasticsearch/xpack/enterprisesearch/action/decrypt/DecryptTransportAction.java

```diff
@@ -0,0 +1,61 @@
+
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.enterprisesearch.action.decrypt;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.enterprisesearch.crypto.CryptoService;
+
+public class DecryptTransportAction extends HandledTransportAction<DecryptRequest, DecryptResponse> {
+
+    private final NodeClient client;
+    private final CryptoService cryptoService;
+    protected Logger logger = LogManager.getLogger(getClass());
+
+    @Inject
+    public DecryptTransportAction(TransportService transportService, ActionFilters actionFilters, NodeClient client) {
+        super(DecryptAction.NAME, transportService, actionFilters, DecryptRequest::new);
+        this.client = client;
+        this.cryptoService = new CryptoService(client.settings());
```

Comment:
You can probably declare CryptoService as a component in the Plugin, and thus get it injected in the constructor without having to instantiate it every time an action is instantiated.